### PR TITLE
Add integration and E2E test for proxy config with username and password

### PR DESCRIPTION
**Pull request by sfc-gh-japatel**
This includes following:
### Commit 1:
- Setup proxy server
- Run integration test and end to end test with proxy setup.
- Proxy username passwords are always admin and test.
- Integration test which is currently disabled but requires a proxy on local machine

### Commit 2
Which was approved here: https://github.com/snowflakedb/snowflake-kafka-connector/pull/273

## Testing:
- Tested locally with newer version of snowpipe SDK jar. 
- Tested with github actions act: https://github.com/nektos/act
- It is able to create connector and also able to ingest a file in snowflake. 

## Prerequisite:
- Waiting for tests to pass in workflow.

### CODE REVIEW

File: .github/workflows/End2EndTestConfluentAws.yml [hunk 1]
```diff
@@ -33,9 +33,23 @@ jobs:
...
+        sudo apt-get -y install squid3
+        sudo apt-get install apache2-utils
+
+    - name: Change squid config and run Proxy Server
+      run: |
+        sudo touch /etc/squid/passwords
+        sudo chmod 777 /etc/squid/passwords
+        sudo htpasswd -db -c /etc/squid/passwords admin test
+        sudo mv .github/scripts/squid.conf /etc/squid/squid.conf
+        sudo service squid start
```
Comment (sfc-gh-zli) [inline .github/workflows/End2EndTestConfluentAws.yml:51]: You also need to do this for those azure workflows.

Comment (sfc-gh-japatel) [reply .github/workflows/End2EndTestConfluentAws.yml:51]: Yes, I will do it after one of this passes.

**Review by sfc-gh-zli — APPROVED** [review]
LGTM

**Comment by sfc-gh-japatel** [comment]
@sfc-gh-zli I merged https://github.com/snowflakedb/snowflake-kafka-connector/pull/273 to this branch.
Will wait for tests to pass and push it to master.

File: src/main/java/com/snowflake/kafka/connector/internal/SnowflakeInternalStage.java [hunk 2]
```diff
@@ -115,6 +121,7 @@ public void putWithCache(String stage, String fullFilePath, String data)
             .setUploadStream(inStream)
             .setRequireCompress(true)
             .setOcspMode(OCSPMode.FAIL_OPEN)
+            .setProxyProperties(proxyProperties)
```
Comment (sfc-gh-japatel) [inline src/main/java/com/snowflake/kafka/connector/internal/SnowflakeInternalStage.java:124]: FYI @sfc-gh-mknister

**Pull request merged**